### ISSUE TITLE
does darknet work well with any image type?

### ISSUE THREAD

**Issue by stephanecharette** [AlexeyAB/darknet#4664]
Looking at some medical and other image datasets, I sometimes see images which are not jpeg.  I have one particular dataset for example which is about 75% jpeg, and the other 25% of the images are a mix of windows bitmap, png, gif, and tiff.

Would this cause problems with Darknet (compiled with OPENCV=1)?

There are two reasons I ask:

1) In one particular case, all the images have a .jpg extension, regardless of the actual image type. 
 I don't know why it was done this way, I can only assume that someone didn't understand that changing the extension doesn't actually change the image type.  Should I rename the files correctly, does it not matter, or should I convert everything to jpeg?

2) Looking at the darknet output during training, I see some errors like this:

```
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 16 Avg (IOU: 0.406856, GIOU: 0.325656), Class: 0.497983, Obj: 0.433956, No Obj: 0.474676, .5R: 0.316667, .75R: 0.016667, count: 60, loss = 168.874786, class_loss = 164.820694, iou_loss = 4.054092
v3 (mse loss, NormaCorrupt JPEG data: 239 extraneous bytes before marker 0xd9
lizer: (iou: 0.750000, cls: 1.000000) Region 23 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.507286, .5R: -nan, .75R: -nan, count: 0, loss = 727.984070, class_loss = 727.984070, iou_loss = 0.000000
```
Note the embedded message about the corrupt JPEG, though it doesn't tell me which image file it is processing when this error is logged.

With max_batches set to 40,000, I see this message show up 7349 times.  Example:

```
Corrupt JPEG data: 239 extraneous bytes before marker 0xd9
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 16 Avg (IOU: 0.786963, GIOU: 0.780216), Class: 0.960369, Obj: 0.504643, No Obj: 0.004255, .5R: 0.981818, .75R: 0.745455, count: 55, loss = 1.196851, class_loss = 0.83525Corrupt JPEG data: 239 extraneous bytes before marker 0xd9
Corrupt JPEG data: 162 extraneous bytes before marker 0xd9
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 23 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.000059, .5R: -nan, .75R: -nan, count: 0, loss = 0.019166, class_loss = 0.0Corrupt JPEG data: 162 extraneous bytes before marker 0xd9
Corrupt JPEG data: 162 extraneous bytes before marker 0xd9
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 16 Avg (IOU: 0.806570, GIOU: 0.800685), Class: 0.965447, Obj: 0.560026, No Obj: 0.006260, .5R: 1.000000, .75R: 0.750000, coCorrupt JPEG data: 162 extraneous bytes before marker 0xd9
Corrupt JPEG data: 239 extraneous bytes before marker 0xd9
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 23 Avg (IOU: 0.700312, GIOU: 0.682440), Class: 0.780498, Obj: 0.139268, No Obj: 0.000984, .5R: 0.972222, .75R: 0.333333, count: 36, loss = 1.601436, class_loss = 1.140975, iou_loss = 0.4Corrupt JPEG data: 239 extraneous bytes before marker 0xd9
```

Thanks for any information you can provide.

**Comment by AlexeyAB** [comment]
Darknet uses OpenCV or STB for loading images, so it supports the most of image types.

For training there are implemented such image to label replaces: https://github.com/AlexeyAB/darknet/blob/2116cba1ed123b38b432d7d8b9a2c235372fd553/src/utils.c#L233-L271
